.travis.yml: Require JupyterHub>0.8.1 and Python>3.5

rkdarst opened the pull request:
- Mirrors the latest requirements of JupyterHub.
- Discussion in #78.

I'm not necessarily saying that this is a good idea. From #78:

I tend to think that HPC is rather conservative and slow to upgrade, so my inclination is to support backwards compatibility a lot. But, JupyterHub is changing so fast it probably doesn't make sense to use anything older than 0.8.1 anyway.

If it was up to me, I would say "yes" for the main reason that I'm not going to be able to support older versions very well since I have only recently started here.  If we don't do this, can someone help with older versions?

Thoughts?

willingc commented:
Good points @rkdarst. I left comments in #78. Perhaps adding versions back in the test matrix and see where that breaks would be a reasonable experiment. ☀️

willingc commented:
And thank you @rkdarst for all of the PRs. I will review the code on all of them today (sorry for the delay). After I do that, perhaps @minrk and @mbmilligan can take a look from an HPC perspective.

rkdarst commented:
Now updated for {0.8, 0.9, master} × {3.5, 3.6} + (0.5, 3.3) + (0.7, 3.4).  I also removed master from allowed_failures: at least for the way I am developing, requiring it to succeed on master is acceptable.

If the progress method can't me made to work on 3.3-3.4 and 3.5-3.6 at the same time, we may have to remove the python<=3.4.

Thoughts?

This is still to-be discussed and not ready to go.

mbmilligan commented:
Ok, this latest commit looks like it matches my recommendations from #78, so I'm happy with that.

Looks like Travis doesn't actually support Python 3.7 yet though?

willingc commented:
Yeah, 3.7 is still in RC so I'm not surprised. Perhaps add nightly but allow failures on it.
